common/src/main/java/io/netty/util/ResourceLeakException.java

```diff
@@ -1,70 +0,0 @@
-/*
- * Copyright 2013 The Netty Project
- *
- * The Netty Project licenses this file to you under the Apache License,
- * version 2.0 (the "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at:
- *
- *   https://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
-
-package io.netty.util;
-
-import java.util.Arrays;
-
-/**
- * @deprecated This class will be removed in the future version.
- */
-@Deprecated
-public class ResourceLeakException extends RuntimeException {
```

Comment:
My 2c: all removals in this PR does not seem like too much of the tech debt and should be ok to carry forward for all 4.x until 5.x. With any removal (even if they are in an "internal" package), you increase the risk of either a new version not being adopted or creating a runtime issues for users (imaging one dependency that upgraded to netty 4.2 and another one that is still on 4.1, it runtime they may not work with each other). 

I would propose to use 4.2 only to rise the min JDK version, while keeping all API as backward compatible.

If you really wanna try removing them, let's try. Just be ready to revert them back in the following 4.2.x versions if it causes issue for users. Taking into account how netty is wide-spread, the likelihood for that to happen is quite high.

Reply:
Code search on GitHub finds no usages of this exception, outside of Netty forks.

The IntegerHolder has a few usages: https://github.com/search?q=%22import+io.netty.util.internal.IntegerHolder%22&type=code

Unfortunately, there has been a fair amount of appetite for the ConcurrentSet: https://github.com/search?q=%22import+io.netty.util.internal.ConcurrentSet%22&type=code

So if we're having second thoughts, then the ConcurrentSet is the most important of the classes to keep.
What do you say, @normanmaurer ?